device/core/src/client_diagnostic.ts

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { Message, errors } from 'azure-iot-common';
+
+/**
+ * Data structure to store diagnostic property data
+ */
+export class DiagnosticPropertyData {
+  // Key of creation time in diagnostic context.
+  static DIAGNOSTIC_CONTEXT_CREATION_TIME_UTC_PROPERTY: string = 'creationtimeutc';
+  // Equal symbol in diagnostic context.
+  static DIAGNOSTIC_CONTEXT_SYMBOL_EQUAL: string = '=';
+  // And symbol in diagnostic context.
+  static DIAGNOSTIC_CONTEXT_SYMBOL_AND: string = '&';
+
+  private diagnosticId: string;
+  private diagnosticCreationTimeUtc: string;
+
+  /**
+   * @constructor
+   */
+  constructor(diagnosticId: string, diagnosticCreationTimeUtc: string) {
+    /* Codes_SRS_DIAGNOSTICPROPERTYDATA_01_001: [If the diagnosticId or diagnosticCreationTimeUtc is null, the constructor shall throw an IllegalArgumentException.] */
+    if (!diagnosticId || !diagnosticCreationTimeUtc) {
+      throw new errors.ArgumentError('The diagnosticId or diagnosticCreationTimeUtc cannot be null or empty.');
+    }
```

Comment:
compiler checks that for us. since this is an internal API, no need for the extra tests.

Reply:
removed null check